Test directory naming simplification introduces collisions

I think the new directory naming convention introduced in #27 is a win, but we need an additional component in the name. Many (all?) of the tests are run in parallel, and the `CoreSimpleTests` and `CoreISATests` end up sharing the same directory for the `rv32ui-p-simple` test, and thus colliding (especially on heavily loaded systems):
```
[info] CoreSimpleTests:
[info] mini.CoreSimpleTests *** ABORTED ***
[info]   java.io.IOException: Cannot run program "./VCoreTester" (in directory "test_run_dir/CoreTester_rv32ui-p-simple"): error=26, Text file busy
```
Jenkins testing shows collisions generating the simulation source files:
```
In file included from VTileTester__ALLcls.cpp:3:0:
VTileTester.cpp:11516:1: error: expected ‘}’ at end of input
 }
 ^
VTileTester.cpp:11516:1: error: expected unqualified-id at end of input
VTileTester.cpp:11516:1: error: expected ‘}’ at end of input
VTileTester.cpp:11516:1: error: expected unqualified-id at end of input
```

--- a/src/test/scala/mini/TestUtils.scala
+++ b/src/test/scala/mini/TestUtils.scala
@@ -224,7 +224,7 @@ abstract class IntegrationTests[T <: BasicTester : ClassTag](
     val subresults = subtests map { test =>
       val stream = getClass.getResourceAsStream(s"/$test.hex")
       val loadmem = io.Source.fromInputStream(stream).getLines
-      Future(test -> (TesterDriver.execute(() => tester(loadmem, testType.maxcycles), nameHint = Some(test))))
+      Future(test -> (TesterDriver.execute(() => tester(loadmem, testType.maxcycles), nameHint = Some(testType.namePrefix + test))))
     }
     Await.result(Future.sequence(subresults), Duration.Inf)
   }

--- a/src/test/scala/mini/Utils.scala
+++ b/src/test/scala/mini/Utils.scala
@@ -5,11 +5,13 @@ package mini
 trait TestType {
   def tests: List[String]
   def maxcycles: Long
+  def namePrefix: String
 }
 
 case object SimpleTests extends TestType {
   val tests = List("rv32ui-p-simple")
   val maxcycles = 15000L
+  val namePrefix = "SimpleTests"
 }
 
 case object ISATests extends TestType {
@@ -21,6 +23,7 @@ case object ISATests extends TestType {
     "sbreak", "scall", "illegal", "ma_fetch", "ma_addr", "csr" //, TODO: "timer"
   ) map (t => s"rv32mi-p-${t}"))
   val maxcycles = 15000L
+  val namePrefix = "ISATests"
 }
 
 case object BmarkTests extends TestType {
@@ -28,6 +31,7 @@ case object BmarkTests extends TestType {
     "median.riscv", "multiply.riscv", "qsort.riscv", "towers.riscv", "vvadd.riscv"
   )
   val maxcycles = 1500000L
+  val namePrefix = "BmarkTests"
 }
 
 case object LargeBmarkTests extends TestType {
@@ -36,4 +40,5 @@ case object LargeBmarkTests extends TestType {
     "towers.riscv-large", "vvadd.riscv-large"
   )
   val maxcycles = 5000000L
+  val namePrefix = "LargeBmarkTests"
 }

